src/events/message.rs

```diff
@@ -59,6 +65,46 @@ pub async fn message(ctx: client::Context, msg: Message) -> Result<()> {
     Ok(())
 }
 
+async fn handle_highlighting(ctx: &client::Context, msg: &Message) -> Result<()> {
+    let db = ctx.get_db().await;
+    let highlights = db.get_highlights().await?;
+
+    for i in &mut msg
+        .content
+        .split_whitespace()
+        .into_iter()
+        .map(|p| p.to_string())
+    {
+        // checks if the database contains our word
+        if let Some(m) = highlights.get(&i.to_string()) {
```

Comment:
you can simplify this to sth along the lines of
```suggestion
    let triggered_highlights = msg
      .content
      .split_whitespace()
      .iter()
      .filter_map(|word| Some((word, highlights.get(word)?)));
    for (word, user_id) in triggered_highlights {
      /......
    }
```